core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -578,58 +580,70 @@ public long getCurrentPageNumber() {
     }
 
     public long spill() throws IOException {
-      synchronized (this) {
-        if (inMemSorter == null) {
-          return 0L;
-        }
-
-        long currentPageNumber = upstream.getCurrentPageNumber();
+      List<MemoryBlock> pagesToFree = Lists.newArrayList();
```

Comment:
Minor nit: Guava says that `Lists.newArrayList` should be treated as deprecated in Java 7+: https://guava.dev/releases/21.0/api/docs/com/google/common/collect/Lists.html#newArrayList--

To avoid us having to update this code in the future, could you change this to 

```
ArrayList<MemoryBlock> pagesToFree = new ArrayList<>();
```

and update the imports?

Reply:
Done, thanks!